pineappl_py/tests/test_grid.py

```diff
@@ -341,6 +341,49 @@ def test_fill(self):
         )
         pytest.approx(res) == 0.0
 
+    def test_fill_array(self):
+        g = self.fake_grid()
+        g.fill_array(
+            order=0,
+            observables=np.array([1e-3, 1e-2]),
+            channel=0,
+            ntuples=[
+                np.array([0.5, 1.0, 1.0]),
+                np.array([0.5, 1.0, 1.0]),
+                np.array([0.5, 1.0, 1.0]),
+            ],
+            weights=np.array([10.0, 100.0]),
+        )
+
+        # Convolution of two symmetrical hadrons
+        h = ConvType(polarized=False, time_like=False)
+        h_conv = Conv(conv_type=h, pid=2212)
+        res = g.convolve(
+            pdg_convs=[h_conv, h_conv],
+            xfxs=[lambda pid, x, q2: x, lambda pid, x, q2: x],
+            alphas=lambda q2: 1.0,
+        )
+        pytest.approx(res) == 0.0
```

Comment:
we should not test against `0.0`, but against something else (as we want to see something is actually inside the grid)

Reply:
Yes, absolutely! These were parts of the legacy checks that were there before. But we'll definitely improve these.

Reply:
Done in [84a4e97](https://github.com/NNPDF/pineappl/pull/299/commits/84a4e9771ba5563e7f9b8edf9e12f3ab1bfb3b18).